### PR TITLE
Document files/directories requiring backmerging to WP Core for major release

### PR DESCRIPTION
**Pull request by getdave**
## What?

Documents the files/paths that do/don't needing back merging to WP Core for a major release.

## Why?

Currently release leads spend a lot of time answering "Should this be backported?" questions. By documenting the basics we relieve some burden on those individuals and make it easier for future release leads.

## How?

- Document the high level scope for files that need back merging
- Document any exceptions

We should also bring `bin/generate-php-sync-issue.mjs` into line with whatever is documented here.

### CODE REVIEW

File: docs/contributors/code/back-merging-to-wp-core.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+The following directories/files do _not_ require back-merging to WP Core:
+
+-   `lib`
+    -   `load.php` - Plugin specific code.
+    -   `experiments-page.php` - experiments are Plugin specific.
+-   `packages`
+    -   `block-library` - this is handled automatically during the packages sync process.
+    -   `e2e-tests/plugins` - PHP files related to e2e tests only. Mostly fixture data generators.
+-   ## `phpunit`
```
Comment (Mamaduka) [inline docs/contributors/code/back-merging-to-wp-core.md]: Does this have to be a header?

**Review by Mamaduka — APPROVED** [review]
Thank you, Dave! I think this is a good start.

File: docs/contributors/code/back-merging-to-wp-core.md (hunk 1 above)
Comment (getdave) [reply docs/contributors/code/back-merging-to-wp-core.md]: It shouldn't be here at all. Tests often require backports.

**Comment by youknowriad** [comment]
nice work here :)

**Comment by getdave** [comment]
Merging and iterating. Any changes that come up in https://github.com/WordPress/gutenberg/issues/57959 should be added to this list.

**Pull request merged**